From 5d06210b3d971c6c7e42235cac25acde6ed8621c Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Mon, 8 Sep 2014 15:07:31 -0400
Subject: [ethernet] ixgbevf: set the disable state when ixgbevf_qv_disable is called

Message-id: <1410188883-6595-17-git-send-email-jogreene@redhat.com>
Patchwork-id: 92341
O-Subject: [RHEL7.1 PATCH 16/48] ixgbevf: set the disable state when ixgbevf_qv_disable is called
Bugzilla: 1091124
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: 1091124
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

The ixgbevf_qv_disable function used by CONFIG_NET_RX_BUSY_POLL is broken,
because it does not properly set the IXGBEVF_QV_STATE_DISABLED bit, indicating
that the q_vector should be disabled (and preventing future locks from
obtaining the vector). This patch corrects the issue by setting the disable
state.

Signed-off-by: Jacob Keller <jacob.e.keller@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit e689e72823e73361147849c416e5da5c4bcc8ddf)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbevf/ixgbevf.h b/drivers/net/ethernet/intel/ixgbevf/ixgbevf.h
index 25c2248..f9cdca9 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf.h
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf.h
@@ -260,6 +260,7 @@ static inline bool ixgbevf_qv_disable(struct ixgbevf_q_vector *q_vector)
  spin_lock_bh(&q_vector->lock);
  if (q_vector->state & IXGBEVF_QV_OWNED)
   rc = false;
+ q_vector->state |= IXGBEVF_QV_STATE_DISABLED;
  spin_unlock_bh(&q_vector->lock);
  return rc;
 }
-- 
1.7.1